### ISSUE TITLE
Unexpected Panic due to empty http response from github

### ISSUE THREAD

**Issue by wileyj** [stacks-network/stacks-blockchain#2990]
**Describe the bug**
`Process abort due to thread panic: panicked at 'Empty response', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/async-h1-1.0.2/src/client.rs:146:9`

**Steps To Reproduce**
intermittent, and dependent on the external service provider

**Expected behavior**
No panic, perhaps an indefinite retry that is logged as to what the issue might be?

**Environment (please complete the following information):**
 - alpine linux
 - stacks-node 2.05.0.0.0 (master:a783b27, release build, linux [x86_64])
 - binaries from official dockerhub image
 - 
**Additional context**
```

Jan 10 09:04:01 AM  INFO [1641834241.102900] [src/net/relay.rs:972] [relayer] Processing newly received Stacks blocks: 6, microblocks: 0, confirmed microblocks: 4
Jan 10 09:04:01 AM  Process abort due to thread panic: panicked at 'Empty response', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/async-h1-1.0.2/src/client.rs:146:9
Jan 10 09:04:02 AM     0: stacks_node::main::{{closure}}
Jan 10 09:04:02 AM               at /src/testnet/stacks-node/src/main.rs:58
Jan 10 09:04:02 AM     1: std::panicking::rust_panic_with_hook
Jan 10 09:04:02 AM               at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panicking.rs:628
Jan 10 09:04:02 AM     2: std::panicking::begin_panic::{{closure}}
Jan 10 09:04:02 AM               at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panicking.rs:544
Jan 10 09:04:02 AM     3: std::sys_common::backtrace::__rust_end_short_backtrace
Jan 10 09:04:02 AM               at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/sys_common/backtrace.rs:141
Jan 10 09:04:02 AM     4: std::panicking::begin_panic
Jan 10 09:04:02 AM               at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panicking.rs:543
Jan 10 09:04:02 AM     5: async_h1::client::decode::{{closure}}
Jan 10 09:04:02 AM               at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/async-h1-1.0.2/src/client.rs:146
Jan 10 09:04:02 AM        <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
Jan 10 09:04:02 AM               at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/core/src/future/mod.rs:80
Jan 10 09:04:02 AM        async_h1::client::connect::{{closure}}
Jan 10 09:04:02 AM               at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/async-h1-1.0.2/src/client.rs:62
Jan 10 09:04:02 AM     6: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
Jan 10 09:04:02 AM               at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/core/src/future/mod.rs:80
Jan 10 09:04:02 AM        stacks_node::event_dispatcher::EventObserver::send_payload::{{closure}}
Jan 10 09:04:02 AM               at /src/testnet/stacks-node/src/event_dispatcher.rs:123
Jan 10 09:04:02 AM        <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
Jan 10 09:04:02 AM               at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/core/src/future/mod.rs:80
Jan 10 09:04:02 AM        async_std::task::block_on::block_on::{{closure}}
Jan 10 09:04:02 AM               at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/async-std-1.5.0/src/task/block_on.rs:68
Jan 10 09:04:02 AM        <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
Jan 10 09:04:02 AM               at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/core/src/future/mod.rs:80
Jan 10 09:04:02 AM     7: async_std::task::block_on::run::{{closure}}
Jan 10 09:04:02 AM               at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/async-std-1.5.0/src/task/block_on.rs:128
Jan 10 09:04:02 AM        std::thread::local::LocalKey<T>::try_with
Jan 10 09:04:02 AM               at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/thread/local.rs:399
Jan 10 09:04:02 AM        std::thread::local::LocalKey<T>::with
Jan 10 09:04:02 AM               at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/thread/local.rs:375
Jan 10 09:04:02 AM     8: async_std::task::block_on::run
Jan 10 09:04:02 AM               at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/async-std-1.5.0/src/task/block_on.rs:118
Jan 10 09:04:02 AM        async_std::task::block_on::block_on::{{closure}}
Jan 10 09:04:02 AM               at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/async-std-1.5.0/src/task/block_on.rs:72
Jan 10 09:04:02 AM        async_std::task::task::Task::set_current::{{closure}}
Jan 10 09:04:02 AM               at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/async-std-1.5.0/src/task/task.rs:129
Jan 10 09:04:02 AM        std::thread::local::LocalKey<T>::try_with
Jan 10 09:04:02 AM               at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/thread/local.rs:399
Jan 10 09:04:02 AM        std::thread::local::LocalKey<T>::with
Jan 10 09:04:02 AM               at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/thread/local.rs:375
Jan 10 09:04:02 AM     9: async_std::task::task::Task::set_current
Jan 10 09:04:02 AM               at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/async-std-1.5.0/src/task/task.rs:124
Jan 10 09:04:02 AM        async_std::task::block_on::block_on
Jan 10 09:04:02 AM               at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/async-std-1.5.0/src/task/block_on.rs:72
Jan 10 09:04:02 AM    10: stacks_node::event_dispatcher::EventObserver::send_payload
Jan 10 09:04:02 AM               at /src/testnet/stacks-node/src/event_dispatcher.rs:114
Jan 10 09:04:02 AM    11: stacks_node::event_dispatcher::EventObserver::send
Jan 10 09:04:02 AM               at /src/testnet/stacks-node/src/event_dispatcher.rs:373
Jan 10 09:04:02 AM        stacks_node::event_dispatcher::EventDispatcher::process_chain_tip
Jan 10 09:04:02 AM               at /src/testnet/stacks-node/src/event_dispatcher.rs:687
Jan 10 09:04:02 AM    12: <stacks_node::event_dispatcher::EventDispatcher as blockstack_lib::chainstate::coordinator::BlockEventDispatcher>::announce_block
Jan 10 09:04:02 AM               at /src/testnet/stacks-node/src/event_dispatcher.rs:437
Jan 10 09:04:02 AM    13: blockstack_lib::chainstate::coordinator::ChainsCoordinator<T,N,U,CE,FE>::process_ready_blocks
Jan 10 09:04:02 AM               at /src/src/chainstate/coordinator/mod.rs:832
Jan 10 09:04:02 AM    14: blockstack_lib::chainstate::coordinator::ChainsCoordinator<T,N,U,CE,FE>::handle_new_stacks_block
Jan 10 09:04:02 AM               at /src/src/chainstate/coordinator/mod.rs:551
Jan 10 09:04:02 AM        blockstack_lib::chainstate::coordinator::ChainsCoordinator<T,blockstack_lib::chainstate::coordinator::comm::ArcCounterCoordinatorNotices,blockstack_lib::chainstate::coordinator::OnChainRewardSetProvider,CE,FE>::run
Jan 10 09:04:02 AM               at /src/src/chainstate/coordinator/mod.rs:316
Jan 10 09:04:02 AM    15: stacks_node::run_loop::neon::RunLoop::start::{{closure}}
Jan 10 09:04:02 AM               at /src/testnet/stacks-node/src/run_loop/neon.rs:309
Jan 10 09:04:02 AM        std::sys_common::backtrace::__rust_begin_short_backtrace
Jan 10 09:04:02 AM               at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/sys_common/backtrace.rs:125
Jan 10 09:04:02 AM    16: std::thread::Builder::spawn_unchecked::{{closure}}::{{closure}}
Jan 10 09:04:02 AM               at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/thread/mod.rs:481
Jan 10 09:04:02 AM        <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
Jan 10 09:04:02 AM               at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/core/src/panic/unwind_safe.rs:271
Jan 10 09:04:02 AM        std::panicking::try::do_call
Jan 10 09:04:02 AM               at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panicking.rs:403
Jan 10 09:04:02 AM        std::panicking::try
Jan 10 09:04:02 AM               at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panicking.rs:367
Jan 10 09:04:02 AM        std::panic::catch_unwind
Jan 10 09:04:02 AM               at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panic.rs:129
Jan 10 09:04:02 AM        std::thread::Builder::spawn_unchecked::{{closure}}
Jan 10 09:04:02 AM               at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/thread/mod.rs:480
Jan 10 09:04:02 AM        core::ops::function::FnOnce::call_once{{vtable.shim}}
Jan 10 09:04:02 AM               at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/core/src/ops/function.rs:227
Jan 10 09:04:02 AM    17: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
Jan 10 09:04:02 AM               at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/alloc/src/boxed.rs:1636
Jan 10 09:04:02 AM        <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
Jan 10 09:04:02 AM               at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/alloc/src/boxed.rs:1636
Jan 10 09:04:02 AM        std::sys::unix::thread::Thread::new::thread_start
Jan 10 09:04:02 AM               at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/sys/unix/thread.rs:106
Jan 10 09:04:02 AM  
Jan 10 09:04:02 AM  Dumping core for pid 12
Jan 10 09:04:03 AM  INFO [1641834243.051327] [testnet/stacks-node/src/main.rs:81] [main] stacks-node 2.05.0.0.0 (master:a783b27, release build, linux [x86_64])
Jan 10 09:04:03 AM  INFO [1641834243.051394] [testnet/stacks-node/src/main.rs:114] [main] Loading config at path /stacks-blockchain/Stacks-mainnet.toml
Jan 10 09:04:03 AM  INFO [1641834243.270940] [testnet/stacks-node/src/run_loop/neon.rs:217] [main] Will run as a Follower node
Jan 10 09:04:03 AM  INFO [1641834243.270972] [testnet/stacks-node/src/run_loop/neon.rs:224] [main] Start syncing Bitcoin headers, feel free to grab a cup of coffee, this can take a while
Jan 10 09:04:03 AM  INFO [1641834243.958108] [src/burnchains/burnchain.rs:1271] [main] Syncing Bitcoin blocks: 8.5% (668376 to 670476 out of 718035)
```

**Comment by jcnelson** [comment]
The panic seems to be originating within `async-h1`, which is a 3rd party HTTP client we use in the `stacks-node` package.  Looks like we're pinned to `1.0`, but this problem was fixed in `2.3.2`: https://github.com/http-rs/async-h1/issues/184

Honestly, I'm in favor of just dropping `async-h1` altogether, along with our usage of `async_std`.  We have an HTTP server and client in `blockstack_lib` already, and the only reason we even put up with `async_std` is because `async-h1` needs it.

**Comment by wileyj** [comment]
That works for me!

**Comment by kantai** [comment]
I think for now, it makes sense to just update the dependency if it's easy.

**Comment by kantai** [comment]
This should be resolved in the next release (2.05.0.1.0)

**Comment by jcnelson** [comment]
#2995 is merged, so I'll go ahead and close this here.
